lib/internal/process/parse_args.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const {
+  ArrayIsArray,
+  ArrayPrototypePush,
+  ArrayPrototypeSlice,
+  Number,
+  SafeSet,
+  StringPrototypeReplace,
+  StringPrototypeSplit,
+  StringPrototypeStartsWith,
+} = primordials;
+const { ERR_INVALID_ARG_TYPE } = require('internal/errors').codes;
+
+/**
+ * Returns an Object representation of command-line arguments to a script.
+ *
+ * Default behavior:
+ * - All arguments are considered "boolean flags"; in the returned
+ *   object, the key is the argument (if present), and the value will be `true`.
+ * - Uses `process.argv.slice(2)`, but can accept an explicit array of strings.
+ * - Argument(s) specified in `opts.expectsValue` will have `string` values
+ *   instead of `true`; the subsequent item in the `argv` array will be consumed
+ *   if a `=` is not used.
+ * - If `=` is present, the value will be boolean if and only if the RHS of the
+ *   string is one of string `true` or `false`
+ * - "Bare" arguments are those which do not begin with a `-` or `--` and those
+ *   after a bare `--`; these will be returned as items of the `_` array
+ * - The `_` array will always be present, even if empty.
+ * - Repeated arguments are combined into an array
+ * @param {string[]} [argv=process.argv.slice(2)] - Array of script arguments as
+ * strings
+ * @param {Object} [options] - Options
+ * @param {string[]|string} [opts.expectsValue] - Argument(s) listed here expect
+ * a value
+ * @example
+ * parseArgs(['--foo', '--bar']) // {foo: true, bar: true, _: []}
+ * parseArgs(['--foo', '-b']) // {foo: true, b: true, _: []}
+ * parseArgs(['---foo']) // {foo: true, _: []}
+ * parseArgs(['--foo=bar']) // {foo: true, _: []}
```

Comment:
Here `foo` is set to true, but below (`parseArgs(['--foo=bar', '--foo=baz']) // {foo: ['bar', 'baz'], _: []}`) it's an array of strings. This is a **recipe for disaster** for end users of the CLI tool. The magic behavior can't return different value types based on the end user input, the developer will not take this into account because it's so unexpected and the end users will spend hours trying to figure out what's wrong.

Reply:
Documentation below is incorrect, you can ignore this comment